src/apps/chifra/pkg/account/state.go

```diff
@@ -0,0 +1,74 @@
+package account
+
+import (
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/base"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/rpc"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/types"
+)
+
+type GetMode int
+
+const (
+	GetBalance GetMode = 1 << iota
+	GetNonce
+	GetCode
+	GetDeployed
+	GetProxy
+	GetType
+)
+
+func GetState(chain string, mode GetMode, address base.Address, blockNumber base.Blknum) (state *types.SimpleEthState, err error) {
+	rpcPayload := []rpc.BatchPayload{
```

Comment:
I love that you're using batch, but have you looked at Multicall? Batch, behind the scenes (as far as I know) just repeatedly calls into the RPC for each call. Mutlicall takes a long string of input data bytes, each with its own call and calls a single time to a smart contract. Not sure which is better, but I've been meaning to look into it.

Reply:
Multicall sounds great, I'll experiment with it